### ISSUE TITLE
psichomics need xxxSJ.out.tab file as input, which is not generated in our pipeline for now

### ISSUE THREAD

**Issue by Rhopala** [cumc/xqtl-pipeline#292]
The file is generated with the option "--sjdbGTFfile /path/to/hg19.gtf" in indexing step of STAR alignment

**Comment by hsun3163** [comment]
To add in more information, the `xxxSJ.out.tab` will only be generated if the GTEX star wrapper don't have an annotation gtf option attached to it, which only needed if `genome index was built w/o annotation` as shown below.
```
if args.annotation_gtf is not None:  # only needed if genome index was built w/o annotation
    cmd += f' --sjdbGTFfile {args.annotation_gtf}'
if args.sjdbFileChrStartEnd is None:
    cmd += ' --twopassMode Basic'
```

However, based on the wording, specifying the gtf even if the index was annotated should not break the pipeline.

__________________________________________

The `{...}.out.tab` file in our index cannot be used because they are not sample-specific and have fewer columns than the one generated by star on the fly.

**Comment by gaow** [comment]
@hsun3163 could you re-read your post above and rephrase it to clarify? Based on what you said above, it sound all you need to do is to add an option to use the gene annotation file in STAR, right?

**Comment by hsun3163** [comment]
> @hsun3163 could you re-read your post above and rephrase it to clarify? Based on what you said above, it sound all you need to do is to add an option to use the gene annotation file in STAR, right?

Yes, which is also what I have done.

**Comment by gaow** [comment]
>  the xxxSJ.out.tab will only be generated if the GTEX star wrapper don't have an annotation gtf

"don't have" or "have"?

**Comment by hsun3163** [comment]
> > the xxxSJ.out.tab will only be generated if the GTEX star wrapper don't have an annotation gtf
> 
> "don't have" or "have"?

Sorry just fix it.

**Comment by Rhopala** [comment]
Update, sorry I just figure out psichomics also require xxxReadsPerGene.out.tab as input, which I check generated by adding option "--quantMode GeneCounts"

**Comment by hsun3163** [comment]
The raise of this ticket is actually due to a misunderstanding of what file are missing. As it turned out. We do have those files, which are located in.

```
/mnt/vast/hpc/csg/xqtl_workflow_testing/rna_quant/output/rnaseq_test
```
